torchvision/transforms/transforms.py

```diff
@@ -527,6 +528,70 @@ def __repr__(self):
         return self.__class__.__name__ + '(p={})'.format(self.p)
 
 
+class RandomPerspective(object):
+    """Performs Perspective transformation of the given PIL Image randomly with a given probability.
+
+    Args:
+        p (float): probability of the image being perspectively transformed. Default value is 0.5
+        interpolation : Default- Image.BICUBIC
+    """
+
+    def __init__(self, interpolation=Image.BICUBIC, p=0.5):
+        self.p = p
+        self.interpolation = interpolation
+
+    def __call__(self, img):
+        """
+        Args:
+            img (PIL Image): Image to be Perspectively transformed.
+
+        Returns:
+            PIL Image: Random perspectivley transformed image.
+        """
+        if not F._is_pil_image(img):
+            raise TypeError('img should be PIL Image. Got {}'.format(type(img)))
+
+        if random.random() < self.p:
+            width, height = img.size
+            coeffs = self.get_params(width, height)
+            return F.perspective(img, coeffs, self.interpolation)
+        return img
+
+    @staticmethod
+    def get_params(width, height):
+        """Get parameters for ``perspective`` for a random perspective transform.
+
+        Args:
+            width : width of the image.
+            height : height of the image.
+        Returns:
+            tuple: 8-tuple (a, b, c, d, e, f, g, h) which contains the coefficients
+                    for a perspective transform.
+        """
+
+        topleft = (random.randint(0, int(width / 2)), random.randint(0, int(height / 2)))
+        topright = (random.randint(int(width / 2), width - 1), random.randint(0, int(height / 2)))
+        botright = (random.randint(int(width / 2), width), random.randint(int(height / 2), height - 1))
+        botleft = (random.randint(0, int(width / 2)), random.randint(int(height / 2), height - 1))
+        matrix = []
+        startpoints = [(0, 0), (width - 1, 0), (width - 1, height - 1), (0, height - 1)]
+        endpoints = [topleft, topright, botright, botleft]
+
+        for p1, p2 in zip(endpoints, startpoints):
+            matrix.append([p1[0], p1[1], 1, 0, 0, 0, -p2[0] * p1[0], -p2[0] * p1[1]])
+            matrix.append([0, 0, 0, p1[0], p1[1], 1, -p2[1] * p1[0], -p2[1] * p1[1]])
+
+        A = numpy.asarray(matrix, dtype=numpy.float)
+        B = numpy.array(startpoints).reshape(8)
+
+        res = numpy.dot(numpy.matmul(numpy.linalg.inv(numpy.matmul(A.T, A)), A.T), B)
```

Comment:
All of the above operations can be converted to PyTorch linalg operations.

Reply:
In fact, `res = torch.gels(B, A)`.

Reply:
@vishwakftw 
Yep, we can. Will change it